n3fit/src/n3fit/backends/keras_backend/internal_state.py

```diff
@@ -13,20 +14,37 @@
 import logging
 import random as rn
 
+import keras
+from keras import backend as K
 import numpy as np
-import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import backend as K
 
 log = logging.getLogger(__name__)
 
+# Prepare Keras-backend dependent functions
+if K.backend() in ("torch", "jax"):
 
-def set_eager(flag=True):
-    """Set eager mode on or off
-    for a very slow but fine grained debugging call this function as early as possible
-    ideally after the first tf import
-    """
-    tf.config.run_functions_eagerly(flag)
+    def set_eager(flag=True):
+        """Pytorch is eager by default"""
+        pass
+
+    def set_threading(threads, core):
+        """Not implemented"""
+        pass
+
+elif K.backend() == "tensorflow":
```

Comment:
Should there be an error for `else`, i.e. if the backend is somehow something else?

Reply:
I can put an `else` for completeness, but Keras will just fail way before arriving here.

Reply:
Okay it might be worth to put a comment explaining what happens if neither condition is met, but I'll leave it up to you